lib/puppet/provider/service/windows.rb

```diff
@@ -125,4 +126,44 @@ def self.instances
     end
     services
   end
+
+  def logonaccount
+    logon_account = @resource[:logonaccount] == 'LocalSystem' ? '.\LocalSystem' : @resource[:logonaccount]
```

Comment:
There should be a method already to take an account name and canonicalize to SID, and then you can just compare with `S-1-32-18`. There should also be a constant for that already.